File: android/src/main/java/com/hhwy/fm_baidu_map/FmBaiduMapPlugin.java
package com.hhwy.fm_baidu_map;

import org.json.JSONObject;

import java.lang.reflect.Method;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;
import io.flutter.plugin.common.PluginRegistry.Registrar;

/** FmBaiduMapPlugin */
public class FmBaiduMapPlugin implements MethodCallHandler {
  private  static  FmBaiduMapPluginImp _imp;
  /** Plugin registration. */
  public static void registerWith(Registrar registrar) {
    final MethodChannel channel = new MethodChannel(registrar.messenger(), "fm_baidu_map");
    channel.setMethodCallHandler(new FmBaiduMapPlugin());
    _imp= new FmBaiduMapPluginImp(registrar);
  }

  @Override
  public void onMethodCall(MethodCall call, Result result) {
    FmToolsBase.onMethodCall(_imp,call,result);
  }
}

File: ios/Classes/FmBaiduMapPlugin.m
#import "FmBaiduMapPlugin.h"
#import "./imp/FmBaiduMapPluginImp.h"
#import "FmToolsBase.h"
@implementation FmBaiduMapPlugin{
}

static FmBaiduMapPluginImp* _imp;

+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"fm_baidu_map"
            binaryMessenger:[registrar messenger]];
  FmBaiduMapPlugin* instance = [[FmBaiduMapPlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
  _imp = [[FmBaiduMapPluginImp alloc] initWithRegist:registrar];
    [FmBaiduMapPluginImp initSDK];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
    if ( [FmToolsBase onMethodCall:_imp method:call.method arg:call.arguments result:result]){
        return;
    }
    result(FlutterMethodNotImplemented);
//  if ([@"getPlatformVersion" isEqualToString:call.method]) {
//    result([@"iOS " stringByAppendingString:[[UIDevice currentDevice] systemVersion]]);
//  } else {
//    result(FlutterMethodNotImplemented);
//  }
}

@end
